### PR TITLE
Use Kraken api to get followers

### LINKED ISSUE #294: question about the number of followers
i have more than 100 followers  and i need to load more than 100 

can i ??  , if i can how 

image

### PR DESCRIPTION
**Pull request by Sorrow81**
# Description

Using the kraken api to grab followers.

```python
    def __do_kraken_request(self, query, response_as_json=True):
        url = f"{API}/{query.strip('/')}"
        response = self.twitch_login.session.get(url)
        logger.debug(
            f"Query: {query}, Status code: {response.status_code}, Content: {response.json()}"
        )
        return response.json() if response_as_json is True else response

    def get_followers(self, limit=100, offset=0):
        followers = []
        query = f"/kraken/users/{self.twitch_login.get_user_id()}/follows/channels?limit={limit}&direction=asc&sortby=login&offset={offset}"
        json_response = self.__do_kraken_request(query)
        followers += [i["channel"]["name"] for i in json_response["follows"]]
        if len(followers) == limit:
            followers += self.get_followers(limit, offset + limit)
        return followers
```

Fixes #294 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

I tested this by launching before and after.

#### Before:
image

#### After:
image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md)
- [x] My changes generate no new warnings
- [x] Any dependent changes have been updated in requirements.txt

### CODE REVIEW

File: TwitchChannelPointsMiner/classes/Twitch.py [hunk 1]
```diff
@@ -155,22 +155,22 @@ def get_channel_id(self, streamer_username):
...
-                and json_response["data"]["personalSections"] != []
-            ):
-                return [
-                    fw["user"]["login"]
-                    for fw in json_response["data"]["personalSections"][0]["items"]
-                    if fw["user"] is not None
-                ]
-        except KeyError:
-            return []
+    def __do_kraken_request(self, query, response_as_json=True):
```
Comment (RakSrinaNa) [inline TwitchChannelPointsMiner/classes/Twitch.py:158]: While this works now, isn't there a better API? This is a legacy API and will be stopped in the upcoming months (https://dev.twitch.tv/docs/v5).

Seems to be replaced with https://dev.twitch.tv/docs/api/reference#get-users-follows

File: TwitchChannelPointsMiner/classes/Twitch.py [hunk 2]
```diff
@@ -155,22 +155,22 @@ def get_channel_id(self, streamer_username):
...
+        url = f"{API}/{query.strip('/')}"
+        response = self.twitch_login.session.get(url)
+        logger.debug(
+            f"Query: {query}, Status code: {response.status_code}, Content: {response.json()}"
+        )
+        return response.json() if response_as_json is True else response
+
+    def get_followers(self, limit=100, offset=0):
+        followers = []
+        query = f"/kraken/users/{self.twitch_login.get_user_id()}/follows/channels?limit={limit}&direction=asc&sortby=login&offset={offset}"
```
Comment (RakSrinaNa) [inline TwitchChannelPointsMiner/classes/Twitch.py:168]: IMO the `/kraken` should be moved in `__do_kraken_request`

**Review by RakSrinaNa — CHANGES_REQUESTED** [review]

File: TwitchChannelPointsMiner/classes/Twitch.py (hunk 2 above)
Comment (Sorrow81) [reply TwitchChannelPointsMiner/classes/Twitch.py:168]: Yes of course like this revision : [https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/tree/0bcd1ac0a05815e4baf947e67038d07921a964bf](url)

File: TwitchChannelPointsMiner/classes/Twitch.py (hunk 1 above)
Comment (Sorrow81) [reply TwitchChannelPointsMiner/classes/Twitch.py:158]: When i want to request helix API on this revision [https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/tree/0bcd1ac0a05815e4baf947e67038d07921a964bf](https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/tree/0bcd1ac0a05815e4baf947e67038d07921a964bf), I get 401 error.

image

I guess we need to change the way we authenticate to the API?

**Pull request closed without merging**